addons/medical_treatment/functions/fnc_checkItems.sqf

```diff
@@ -16,27 +16,22 @@ if (EGVAR(medical,convertItems) == 2) exitWith {};
 
 params ["_unit"];
 
-private _countFirstAidKit = [_unit, "FirstAidKit"] call EFUNC(common,getCountOfItem);
-_unit removeItems "FirstAidKit";
+// Item conversions are cached at preStart
+private _itemHash = uiNamespace getVariable QGVAR(replaceItems);
 
-private _countMedikit = [_unit, "Medikit"] call EFUNC(common,getCountOfItem);
-_unit removeItems "Medikit";
+private _replaceItem = {
+    private _itemCount = [_unit, _key] call EFUNC(common,getCountOfItem);
+    _unit removeItems _key;
```

Comment:
No need if _itemCount is 0.
Don't know how expensive it is. You could try packing a backpack full with hundreds of ace bandages,
and running `player removeItems "FirstAidKit"` in debug console to see how expensive it is to remove a item that player doesn't even have.
Knowing Arma, I would assume it iterates over all items in inventory and grabs the ones that match. Will will be hella expensive with 200 items in inventory and trying to remove an item that doesn't exist.